src/identity-providers/add/DescriptorSettings.tsx

```diff
@@ -0,0 +1,428 @@
+                  }
+                  isOpen={samlSignatureKeyNameDropdownOpen}
+                  onSelect={(_, value) => {
+                    onChange(value as string);
+                    setSamlSignatureKeyNameDropdownOpen(false);
+                  }}
+                  selections={value}
+                  variant={SelectVariant.single}
+                >
+                  <SelectOption key={0} value="NONE" />
+                  <SelectOption key={1} value={t("keyID")} isPlaceholder />
+                  <SelectOption key={2} value={t("certSubject")} />
+                </Select>
+              )}
+            ></Controller>
+          </FormGroup>
+        </>
+      )}
+
+      <SwitchField
+        field="config.wantAssertionsSigned"
+        label="wantAssertionsSigned"
+        isReadOnly={readOnly}
+      />
+
+      <SwitchField
+        field="config.wantAssertionsEncrypted"
+        label="wantAssertionsEncrypted"
+        isReadOnly={readOnly}
+      />
+      <SwitchField
+        field="config.forceAuthn"
+        label="forceAuthentication"
+        isReadOnly={readOnly}
+      />
+
+      <SwitchField
+        field="config.validateSignature"
+        label="validateSignature"
+        isReadOnly={readOnly}
+      />
+      {validateSignature === "true" && (
+        <>
+          <TextField
+            field="config.signingCertificate"
+            label="validatingX509Certs"
+            isReadOnly={readOnly}
+          />
+        </>
+      )}
+      <SwitchField
+        field="config.signSpMetadata"
+        label="signServiceProviderMetadata"
+        isReadOnly={readOnly}
+      />
+      <SwitchField
+        field="config.passSubject"
+        label="passSubject"
+        isReadOnly={readOnly}
+      />
+
+      <FormGroup
+        label={t("allowedClockSkew")}
+        labelIcon={
+          <HelpItem
+            helpText={th("allowedClockSkew")}
+            forLabel={t("allowedClockSkew")}
+            forID="allowedClockSkew"
+          />
+        }
+        fieldId="allowedClockSkew"
+        helperTextInvalid={t("common:required")}
+      >
+        <TextInput
+          type="text"
+          id="allowedClockSkew"
+          name="config.allowedClockSkew"
+          isReadOnly={readOnly}
+        />
+      </FormGroup>
+    </div>
+  );
+};
+
+export const DescriptorSettings = ({ readOnly }: DescriptorSettingsProps) => {
+  const { t } = useTranslation("identity-providers");
+  const [isExpanded, setIsExpanded] = useState(false);
+
+  return (
+    <>
+      {readOnly && (
+        <ExpandableSection
+          className="keycloak__discovery-settings__metadata"
+          toggleText={isExpanded ? t("hideMetaData") : t("showMetaData")}
+          onToggle={() => setIsExpanded(!isExpanded)}
+          isExpanded={isExpanded}
+        >
+          <Fields readOnly={readOnly} />
+        </ExpandableSection>
+      )}
+      {!readOnly && <Fields readOnly={readOnly} />}
+    </>
+  );
```

Comment:
This could be a ternary expression, which also eliminates the need to use a fragment here.
```suggestion
  return readOnly ? (
    <ExpandableSection
      className="keycloak__discovery-settings__metadata"
      toggleText={isExpanded ? t("hideMetaData") : t("showMetaData")}
      onToggle={() => setIsExpanded(!isExpanded)}
      isExpanded={isExpanded}
    >
      <Fields readOnly={readOnly} />
    </ExpandableSection>
  ) : (
    <Fields readOnly={readOnly} />
  );
```